server/service/src/stock_line/historical_stock.rs

```diff
@@ -38,28 +39,46 @@ pub fn get_historical_stock_lines_available_quantity(
     stock_movements.sort_by(|a, b| b.datetime.cmp(&a.datetime));
 
     // Calculate available stock for each stock line currently
-    let mut min_available_and_pack_size: HashMap<String, MinAvailableAndPackSize> = stock_lines
-        .iter()
-        .map(|(stock_line, reserved_available_number_of_packs)| {
-            let total = stock_line.total_number_of_packs * stock_line.pack_size;
-            let available_packs = stock_line.available_number_of_packs
-                + reserved_available_number_of_packs.unwrap_or_default();
-            let available = available_packs * stock_line.pack_size;
-            (
-                stock_line.id.clone(),
-                MinAvailableAndPackSize {
-                    pack_size: stock_line.pack_size,
-                    min: available,
-                    total,
-                },
-            )
-        })
-        .collect();
+    let mut min_available_and_pack_size: HashMap<String, MinAvailableAndPackSize> =
+        stock_lines_plus_reserved
+            .iter()
+            .map(|(stock_line, reserved_available_number_of_packs)| {
+                // Any stock already allocated to this invoice, will have reduced the available stock in stock lines.
```

Comment:
Added comments here as it wasn't obvious the intention was to undo the currently allocated invoice lines to correct the total.